app/models/ApplicationRequest.scala

```diff
@@ -56,67 +66,124 @@ case class EORIDetails(
   postcode: String,
   country: String
 )
+object EORIDetails {
+  implicit val format: OFormat[EORIDetails] = Json.format[EORIDetails]
+}
 
 case class ContactDetails(
   name: String,
   email: String,
   phone: Option[String]
 )
+object ContactDetails {
+  implicit val format: OFormat[ContactDetails] = Json.format[ContactDetails]
+}
 
-/** determined via `_type` field, e.g. "_type" = "MethodOne" { "_type": "MethodOne",
-  * "saleBetweenRelatedParties": "blah blah", ...
-  */
 sealed trait RequestedMethod
 
 case class MethodOne(
   saleBetweenRelatedParties: Option[String],
   goodsRestrictions: Option[String],
   saleConditions: Option[String]
 ) extends RequestedMethod
+object MethodOne {
+  implicit val format: OFormat[MethodOne] = Json.format[MethodOne]
+}
 
-sealed trait IdenticalGoodsExplaination
-// again, "_type": "PreviouslyImportedGoods", etc
-case class PreviousIdenticalGoods(value: String) extends IdenticalGoodsExplaination
-case class OtherUsersIdenticalGoods(value: String) extends IdenticalGoodsExplaination
+sealed trait IdenticalGoodsExplaination extends Any
+case class PreviousIdenticalGoods(val value: String) extends AnyVal with IdenticalGoodsExplaination
+object PreviousIdenticalGoods {
+  implicit val format: Format[PreviousIdenticalGoods] = Json.valueFormat[PreviousIdenticalGoods]
+}
+case class OtherUsersIdenticalGoods(val value: String)
+    extends AnyVal
+    with IdenticalGoodsExplaination
+object OtherUsersIdenticalGoods {
+  implicit val format = Json.valueFormat[OtherUsersIdenticalGoods]
+}
+
+object IdenticalGoodsExplaination {
+  implicit val format: Format[IdenticalGoodsExplaination] = Json.format[IdenticalGoodsExplaination]
+}
 
 case class MethodTwo(
   whyNotOtherMethods: String,
   detailedDescription: IdenticalGoodsExplaination
 ) extends RequestedMethod
+object MethodTwo {
+  implicit val format: OFormat[MethodTwo] = Json.format[MethodTwo]
+}
 
-sealed trait SimilarGoodsExplaination
-// again, "_type": "PreviouslyImportedGoods", etc
-case class PreviousSimilarGoods(value: String) extends SimilarGoodsExplaination
-case class OtherUsersSimilarGoods(value: String) extends SimilarGoodsExplaination
+sealed trait SimilarGoodsExplaination extends Any
+case class PreviousSimilarGoods(val value: String) extends AnyVal with SimilarGoodsExplaination
+case class OtherUsersSimilarGoods(val value: String) extends AnyVal with SimilarGoodsExplaination
+
+object PreviousSimilarGoods {
+  implicit val format: Format[PreviousSimilarGoods] = Json.valueFormat[PreviousSimilarGoods]
+}
+object OtherUsersSimilarGoods {
+  implicit val format: Format[OtherUsersSimilarGoods] = Json.valueFormat[OtherUsersSimilarGoods]
+}
+
+object SimilarGoodsExplaination {
```

Comment:
This provides an encoder/decoder for a sealed trait. I wonder if it is unrequired considering that all the inheritors of the trait already have encoders.